src/client/opamPinCommand.ml

```diff
@@ -474,6 +475,16 @@ and source_pin
       OpamSwitchState.opam_opt st nv)
   in
 
+  let opam_opt =
+    match opam_local, opam_opt with
+    | Some local, None ->
+      OpamConsole.warning
+        "Couldn't retrieve opam file from versionned source, \
+         using the one found locally.";
+      Some local
```

Comment:
I am wondering if the case when both are defined and different (`Some local, Some vers when local <> vers`; or maybe `when not (OpamFile.OPAM.effectively_equal local vers)`) would not deserve a warning too: "not using uncommitted version of opam file xxx".

But there is already the warning about uncommitted changes, so that might be redundant.

Reply:
@AltGr It would be complementary: on update, uncommitted changes warning is shown, in pinning, uncommitted opam changes warning is shown.